tests/test_axis.py

```diff
@@ -820,9 +849,13 @@ def test_init(self):
         with pytest.raises(TypeError):
             bh.axis.Boolean(1)
 
+    def test_traits(self):
         ax = bh.axis.Boolean()
         assert isinstance(ax, bh.axis.Boolean)
-        assert ax.options == bh.axis.options()
+
+        # Note: this axis not inclusive, since we don't have a dedicated bool
```

Comment:
This is bad, the Boolean axis should behave like the Boost.Histogram one, it should be inclusive. One of the reasons to use an explicit Boolean axis is to safe memory when there are really only two alternatives. It does not matter that there is no bool fill. A fill with integers will be cast to boolean values, which is as it should be when you explicitly make a Boolean axis.

Reply:
I think we decided to make the Boolean axis just a wrapper over the Integer axis in Python, but it should still behave the same way as in Boost.Histogram.

Reply:
It does behave the same way - it is "inclusive" of bools - but since it's an integer axis to avoid adding 50% extra compile complexity on things like `.fill`, it's not "truly" inclusive (as in, not all integers). We need this to not be a true Bool axis from Boost.Histogram because all axis are sliceable - if someone wants to do `h[0:1]`, that's valid in boost-histogram, it's a false-only axis. You helped me come up with this axis type, IIRC.

Now, we could override `.inclusive` on our custom Bool axis, I believe - not knowing what optimizations are done here, I didn't change that - but maybe that's actually better. Inclusive would be "true" if length 2, and "false" if this was length 1.